### PR TITLE
docs($window): improve the example

### PR DESCRIPTION
**Pull request by thorn0**

### CODE REVIEW

File: src/ng/window.js [hunk 1]
```diff
@@ -22,22 +22,37 @@
          angular.module('windowExample', [])
            .controller('ExampleController', ['$scope', '$window', function($scope, $window) {
              $scope.greeting = 'Hello, World!';
-             $scope.doGreeting = function(greeting) {
-               $window.alert(greeting);
+             $scope.doGreeting = function() {
+               $window.alert($scope.greeting);
+               $scope.greeting = '';
```
Comment (gkalpak) [inline src/ng/window.js]: What's the purpose of this line ?

File: src/ng/window.js [hunk 2]
```diff
@@ -22,22 +22,37 @@
...
+       beforeEach(module(function($provide) {
+         // Mock $window
+         $provide.decorator('$window', function($delegate) {
+           // Use the original window as a prototype for the mock in end-to-end
+           // tests so that the parts of Angular that interact with the browser
+           // via $window won't break.
+           var decoratedWindow = Object.create($delegate);
+           // The browser's implementation of the alert function blocks the test
+           // runner, so replace it in the mock.
+           decoratedWindow.alert = this.alert = jasmine.createSpy('alert');
```
Comment (gkalpak) [inline src/ng/window.js]: I am not keen on `this.alert`.

File: src/ng/window.js [hunk 3]
```diff
@@ -22,22 +22,37 @@
...
+           return decoratedWindow;
+         });
+       }));
+       it('should clear the input field after alert', function() {
+         element(by.model('greeting')).sendKeys('Hello, E2E Tests');
+         element(':button').click();
+         expect(this.alert).toHaveBeenCalled();
```
Comment (gkalpak) [inline src/ng/window.js]: Why not inject `$window` and check up on `$window.alert` instead of having `this.alert`.

File: src/ng/window.js (hunk 2 above)
Comment (thorn0) [inline src/ng/window.js]: Oops. Yes, in that place `this` doesn't mean Jasmine's `this`...

File: src/ng/window.js (hunk 1 above)
Comment (thorn0) [inline src/ng/window.js]: Just some logic to test it in the test.

**Pull request closed without merging**